app/controllers/YourBillController.scala

```diff
@@ -101,11 +132,18 @@ object YourBillController {
   }
 
   private def overDuePaymentOf(ass: ChargeTypeAssessment): OverduePayment =
-    OverduePayment(invoicePeriod(ass), ass.debtTotalAmount.value, chargeBearsInterest(ass))
+    OverduePayment(invoicePeriod(ass), ass.debtTotalAmount.value, chargeBearsInterest(ass), ddInProgress(ass))
+
+  private def qualifyingDebt(eligibilityResult: EligibilityCheckResult): AmountInPence =
+    eligibilityResult.chargeTypeAssessment.map(_.debtTotalAmount.value).fold(AmountInPence.zero)(_ + _)
 
   private def overDuePayments(eligibilityResult: EligibilityCheckResult): OverDuePayments = {
-    val qualifyingDebt: AmountInPence = eligibilityResult.chargeTypeAssessment.map(_.debtTotalAmount.value).fold(AmountInPence.zero)(_ + _)
     val payments = eligibilityResult.chargeTypeAssessment.map(overDuePaymentOf)
-    OverDuePayments(qualifyingDebt, payments)
+    OverDuePayments(qualifyingDebt(eligibilityResult), payments)
+  }
+
+  private def overDuePaymentsWithDdInProgress(eligibilityResult: EligibilityCheckResult): OverDuePayments = {
+    val paymentsWithDdInProgress = eligibilityResult.chargeTypeAssessment.map(overDuePaymentOf).filter(_.ddInProgress.contains(DdInProgress(true)))
```

Comment:
not super-important but the filter can be reduced a bit with something like `.filter(_.ddInProgress.value)` or `.filter(_.ddInProgress)` - I think you already created an implicit conversion to a boolean to make the second syntax work there